src/containers/deactivateAccount/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { Link as RouterLink, Redirect } from 'react-router-dom';
+import { UserAuthenticationReducerState } from '../../auth/ducks/types';
+import { ContentContainer } from '../../components';
+import { C4CState } from '../../store';
+import {
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+  asyncRequestIsLoading,
+} from '../../utils/asyncRequest';
+import { requestToDeactivateAccount } from './ducks/thunks';
+import { DeactivateAccountReducerState } from './ducks/types';
+const { Title, Link } = Typography;
+
+interface DeactivateAccountProps {
+  deactivateAccount: DeactivateAccountReducerState['deactivateAccount'];
+  tokens: UserAuthenticationReducerState['tokens'];
+}
+
+const DeactivateAccount: React.FC<DeactivateAccountProps> = ({
+  deactivateAccount,
+  tokens,
+}) => {
+  const dispatch = useDispatch();
+  const onFinish = () => {
+    dispatch(requestToDeactivateAccount());
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Deactivate Account</title>
+        <meta name="description" content="Permanently delete your account." />
+      </Helmet>
+      <ContentContainer>
+        <Title>Deactivate Account</Title>
+        {(() => {
+          if (asyncRequestIsComplete(deactivateAccount)) {
+            return (
+              <>
+                <Title level={3}>Account Successfully Deactivated</Title>
+                <RouterLink to="/">
+                  <Link>Back to the home page</Link>
+                </RouterLink>
+              </>
+            );
+          } else if (asyncRequestIsComplete(tokens)) {
+            return (
+              <>
+                <Title level={3}>
+                  Warning: This action is permanent and cannot be undone
+                </Title>
+                <Form onFinish={onFinish}>
+                  <Form.Item
+                    name="deactivateAccount"
+                    valuePropName="checked"
+                    rules={[
+                      {
+                        required: true,
+                        type: 'enum',
+                        enum: [true],
+                        message:
+                          'Please check this box to confirm account deletion',
+                      },
+                    ]}
+                  >
+                    <Checkbox>
+                      By checking this box you acknowledge that your information
+                      will be erased from our system and cannot be recovered
+                      including any registrations for future events or requests
+                      for Participating Family status.
+                    </Checkbox>
+                  </Form.Item>
+                  {asyncRequestIsFailed(deactivateAccount) && (
+                    <Alert
+                      message="Error"
+                      description={deactivateAccount.error}
+                      type="error"
+                      showIcon
+                    />
+                  )}
+
+                  <Form.Item>
```

Comment:
Looking at the screenshot, it might be nice to vertically center the deactivate my account button. Possibly adding some padding-top so its not so close to the error message